### PR TITLE
Upgrade to Scala 2.12.0

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -120,7 +120,7 @@ object Build {
   }
 
   val commonSettings = Seq(
-      scalaVersion := "2.11.8",
+      scalaVersion := "2.12.0",
```
Comment (sjrd) [inline project/Build.scala]: IMO we should wait for IDEs to pick up support for 2.12 in release builds before making that specific jump.

**Comment by sjrd** [comment]
Probably worth rebasing on top of master since #2641 was merged.

**Comment by gzm0** [comment]
Updated

**Comment by gzm0** [comment]
Updated (JUnit test white-/blacklists were missing).

**Comment by sjrd** [comment]
Nightly: https://scala-webapps.epfl.ch/jenkins/job/scalajs-matrix-build/4579/

**Comment by sjrd** [comment]
Nightly failed its 2.10.5 job, as expected, but otherwise succeeded.

**Pull request merged**